File: frontend/src/types/watchAlert.ts
export type WatchAlertTriggerType =
  | 'rating_streak'
  | 'goal_streak'
  | 'assist_streak'
  | 'recent_transfer'
  | 'market_value_spike'

export interface WatchAlertPlayer {
  id: number
  full_name: string
  photo_url: string | null
  current_team: string | null
  league: string | null
}

export interface WatchAlert {
  id: number
  player_id: number
  player: WatchAlertPlayer
  // null = segnalazione manuale (vedi is_manual).
  trigger_type: WatchAlertTriggerType | null
  trigger_detail: string
  detected_at: string
  is_dismissed: boolean
  is_manual: boolean
  is_seen: boolean
}

File: backend/app/schemas/watch_alert.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class WatchAlertPlayerOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    full_name: str
    photo_url: str | None
    current_team: str | None
    league: str | None


class WatchAlertOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    player_id: int
    player: WatchAlertPlayerOut
    # None = segnalazione manuale (vedi is_manual).
    trigger_type: str | None
    trigger_detail: str
    detected_at: datetime
    is_dismissed: bool
    is_manual: bool
    is_seen: bool


class ManualWatchAlertCreate(BaseModel):
    player_id: int
    note: str = Field(min_length=1, max_length=1000)

    @property
    def clean_note(self) -> str:
        return self.note.strip()


class WatchAlertUnseenCount(BaseModel):
    count: int
